src/components/MultiSelect/MultiSelectMenuButton.tsx

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import Button from "./../Button/Button";
+import Icon from "./../Icons/Icon";
+import { Box, useMultiStyleConfig, keyframes } from "@chakra-ui/react";
+import { SelectedItems } from "./MultiSelect";
+
+export interface MultiSelectMenuButtonProps {
+  id: string;
+  /** The id of the MultiSelect using this button. */
+  multiSelectId: string;
+  /** The label of the MultiSelect using this button. */
+  multiSelectLabel: string;
+  /** The open status of the MultiSelect menu. */
+  isOpen: boolean;
+  /** The selected items state (items that were checked by user). */
+  selectedItems: SelectedItems;
+  /** The callback function for the menu toggle. */
+  onMenuToggle?: () => void;
+  /** The action to perform for clear/reset button of MultiSelect. */
+  onClear?: () => void;
+  onKeyDown?: () => void;
+}
+const grow = keyframes`
+  from {width: 22px; opacity: 0; }
+  to {width: 46px; opacity: 1;}
+`;
+const MultiSelectMenuButton = React.forwardRef<
+  HTMLButtonElement,
+  MultiSelectMenuButtonProps
+>((props, ref?) => {
+  const {
+    id,
+    multiSelectId,
+    multiSelectLabel,
+    isOpen,
+    onMenuToggle,
+    selectedItems,
+    onClear,
+    onKeyDown,
+    ...rest
+  } = props;
+  const styles = useMultiStyleConfig("MultiSelectMenuButton", {});
+  const iconType = isOpen ? "minus" : "plus";
+  const growAnimation = `${grow} 0.075s ease-out`;
+
+  // Sets the selected items count on the menu button.
+  let getSelectedItemsCount;
+  if (selectedItems[multiSelectId]?.items.length > 0) {
+    getSelectedItemsCount = `${selectedItems[multiSelectId].items.length}`;
+  }
+  // We need this for our "fake" button inside the main menu button.
+  function onKeyPress(e) {
+    const enterOrSpace =
+      e.key === "Enter" ||
+      e.key === " " ||
+      e.key === "Spacebar" ||
+      e.which === 13 ||
+      e.which === 32;
+
+    if (enterOrSpace) {
+      e.preventDefault();
+      onClear();
+    }
+  }
+
+  return (
+    <Button
+      id={id}
```

Comment:
Nit: for all components, organize the props in alphabetical order and put `__css` next to last and `{...rest}` as the last prop.

Reply:
fixed